### ISSUE TITLE
HPA scales up when it is expected to scale down.

### ISSUE THREAD

**Issue by EmanekaT** [kubernetes/kubernetes#83698]
**What happened**:
```
Name:               test
Namespace:          test
Labels:             <none>
Annotations:        kubectl.kubernetes.io/last-applied-configuration:
                      {"apiVersion":"autoscaling/v2beta2","kind":"HorizontalPodAutoscaler","metadata":{"annotations":{"signalfx.com.custom.metrics":""},"name":"...
                    signalfx.com.custom.metrics:
CreationTimestamp:  Wed, 09 Oct 2019 15:49:21 -0700
Reference:          Deployment/test
Metrics:            ( current / target )
  "uwsgi" on pods:  5128m / 53
Min replicas:       1
Max replicas:       3
Deployment pods:    3 current / 3 desired
Conditions:
  Type            Status  Reason               Message
  ----            ------  ------               -------
  AbleToScale     True    ScaleDownStabilized  recent recommendations were higher than current one, applying the highest recent recommendation
  ScalingActive   True    ValidMetricFound     the HPA was able to successfully calculate a replica count from pods metric uwsgi
  ScalingLimited  True    TooManyReplicas      the desired replica count is more than the maximum replica count
Events:
  Type     Reason                        Age   From                       Message
  ----     ------                        ----  ----                       -------
  Warning  FailedGetPodsMetric           104s  horizontal-pod-autoscaler  unable to get metric uwsgi: no metrics returned from custom metrics API
  Warning  FailedComputeMetricsReplicas  104s  horizontal-pod-autoscaler  Invalid metrics (1 invalid out of 1), last error was: failed to get object metric value: unable to get metric uwsgi: no metrics returned from custom metrics API
```
```
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
metadata:
  annotations:
    signalfx.com.custom.metrics: ""
  name: test
  namespace: test
spec:
  scaleTargetRef:
    apiVersion: extensions/v1beta1
    kind: Deployment
    name: test
  minReplicas: 1
  maxReplicas: 3
  metrics:
  - type: Pods
    pods:
      metric:
        name: uwsgi
      target:
        type: AverageValue
        averageValue: 53
```
```
Replicas:               20 desired | 20 updated | 37 total | 1 available | 36 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  19% max unavailable, 100% max surge
```
HPA scales up until it hits maxReplica.  2 out of 3 pods are running but not ready.
I thought it could be due to missing metrics but the behavior is the same without missing metrics. It behaves the same way even if I set metric value to be 0 for unready pods. 

**What you expected to happen**:
Since the usage ratio is 5128m / 53, I expect it to scale down. 

**How to reproduce it (as minimally and precisely as possible)**:
Use the yaml file definition I pasted and run an app to post metrics.

**Anything else we need to know?**:
I wanted to check logs but I only found few from horizontal.go from controller-manager.
I noticed some similar issues (#78761) but it seems some of the fixes should be released to 1.15 already?  

**Environment**:
- Kubernetes version (use `kubectl version`):
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:50Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration:  AWS ec2 
- OS (e.g: `cat /etc/os-release`):
NAME="Ubuntu"
VERSION="16.04.5 LTS (Xenial Xerus)"
- Kernel (e.g. `uname -a`):
Linux kubestage1-uswest1adevc 4.4.0-1085-aws #96-Ubuntu SMP Tue Jun 11 09:08:32 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by EmanekaT** [comment]
Oh it seems to be the same issue as #78761

**Comment by EmanekaT** [comment]
I checked that v1.15.3 does not have the hpa fixes. I'll upgrade my k8s to the lastest version and try again.

**Comment by EmanekaT** [comment]
I confirmed that this is fixed in 1.16.2.
